### ISSUE TITLE
App CPU usage

### ISSUE THREAD

**Issue by beaucollins** [mattrubin/authenticator#218]
When the app is at rest running under a debugger with 11 tokens it uses around 40% CPU

<img width="942" alt="sans titre" src="https://user-images.githubusercontent.com/19795/32126767-47a21ac6-bb27-11e7-8905-52427c3ed3b0.png">

**Comment by beaucollins** [comment]
The `TokenRowModel`'s call to `token.generator.password(at: Date)` takes up about half of that work. By changing it to an empty string CPU usage drops by 20%:

<img width="987" alt="sans titre 2" src="https://user-images.githubusercontent.com/19795/32126905-118ce370-bb28-11e7-92f5-94c1e3dad1dc.png">

Timer based tokens change based on a time period, can we get some performance enhancements by memoizing by time period?

**Comment by beaucollins** [comment]
Tying View Model updates to `CADisplayLink` means the view model is generated every 16⅔ ms (`CADisplayLink` tries to maintain a 60 fps refresh rate).

In iOS 10.0 we can provide a `preferredFramesPerSecond`:

```swift
if #available(iOS 10.0, *) {
  self.displayLink?.preferredFramesPerSecond = 1
}
```

Which drops CPU time down to 1%:

<img width="968" alt="sans titre 3" src="https://user-images.githubusercontent.com/19795/32127118-76d23b44-bb29-11e7-8753-60c1475daf22.png">

The downside is the animated progress circle now only updates once per second. The ring itself could be animated between progress points instead of relying on `CADisplayLink` updates for its animation.
